### ISSUE TITLE
DragFloat: wrong behavior with power != 1

### ISSUE THREAD

**Issue by BrutPitt** [ocornut/imgui#2174]
**Version/Branch of Dear ImGui:**

1.65 and 1.66 WIP (1/Nov/2018)
(No problem with 1.60)

**Back-end file/Renderer/OS:**

Back-ends: imgui_impl_GLFW.cpp + imgui_impl_opengl3.cpp
Renderer: OpenGL (NVidia/AMD/Intel)

OS: Windows/Linux/OSX/Emscripten

**My Issue/Question:** 

When Using DragFloat, with power != 1.0, I have this strange behavior:
After I Drag mouse in a direction, when I change verso, the previous increment or decrement continue as if I had not changed direction, until I do not reach the initial drag point, only at that point also it change, and behaves correctly.
In the example below (animated gif), I drag to the right and the widget correctly increment, but when I change direction, and it should decrease, instead it continues to increase.
This happens ONLY with power != 1
Note: If I reach the upper/lower limit and after I change direction, the problem do not appear.

**Standalone, minimal, complete and verifiable example:** 
Any use of DragFloat with power != 1 I get same behavior, in this example:
```
 ImGui::DragFloat("##Specular Exp", &theWnd->qjSet->specularExponent,.1, 1.0, 250.0,"%.3f",2.f);

 ```
**Screenshots/Video**

![dragfloat](https://user-images.githubusercontent.com/16171743/47979847-6572c680-e0c5-11e8-8b25-76dcbbdb7396.gif)

If prefer you can test it live in emscripten example:
(only for Chrome/Firefox/Opera with WebGL2 support)
[imGuIZMO example](http://www.michelemorrone.eu/caoticheidee/imGuIZMO/qjSet.html)
(Is the _Specular Exp_ Widget, as in animated gif)

**Comment by ocornut** [comment]
Thank you @BrutPitt , this is now fixed by #2206.

**Comment by ocornut** [comment]
Power!=1 is generally quite weird/broken at the moment.
We probably need a change like #1316 at some point.
